drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -1114,6 +1157,30 @@ local function info_changed(driver, device, event, args)
       device:set_field(DEFERRED_CONFIGURE, nil)
     end
   end
+
+  if not device.preferences or device.network_type == device_lib.NETWORK_TYPE_CHILD then
+    return
+  end
+
+  if device.manufacturer_info.vendor_id == fingerprint_profile_overrides[1].vendor_id and
+    device.manufacturer_info.product_id == fingerprint_profile_overrides[1].product_id then
+    local time_diff = 3
+    local last_clock_set_time = device:get_field(LATEST_CLOCK_SET_TIMESTAMP)
+    if last_clock_set_time ~= nil then
+      time_diff = os.difftime(os.time(), last_clock_set_time)
+    end
+    device:set_field(LATEST_CLOCK_SET_TIMESTAMP, os.time(), {persist = true})
+    if time_diff > 2 then
+      local preferences = preference_map_inovelli_vtm31sn
+      for id, value in pairs(device.preferences) do
+        if args.old_st_store.preferences[id] ~= value and preferences and preferences[id] then
+          local new_parameter_value = preferences_to_numeric_value(device.preferences[id])
+          local req = clusters.ModeSelect.server.commands.ChangeToMode(device, preferences[id].parameter_number, new_parameter_value)
```

Comment:
If the mode cluster is going to be used here, this would be the first use of it in the `matter-switch` driver. This means we'd need to add it as an embedded cluster so it could be used on older FW that doesn't support the mode cluster via the lua libs definitions yet.

Reply:
It looks like ModeSelect has been been in the spec since matter 1.0. Is there a way to check when it was first added to the lua libs?